frontend/test/pytest/test_gradient.py

```diff
@@ -1620,13 +1623,22 @@ def body(i, result_array):
 
         return body(transposed_result)
 
-    cat_res = qjit(
+    cat_res_func = qjit(
         jacobian(
             my_model,
             argnums=1,
         ),
         keep_intermediate=True,
-    )(data, params["weights"])
+    )
+
+    # Since the above compilation fails and this test is xfail, the intermediate
+    # will not be properly cleaned up, so we delete it manually
+    try:
+        cat_res = cat_res_func(data, params["weights"])
+    finally:
+        if os.path.isdir("grad.my_model"):
+            shutil.rmtree("grad.my_model")
+
```

Comment:
we already have a utility for this.

Reply:
It is `qfuncname.workspace.cleanup()`

Reply:
Why don't we just remove keep_intermediate=True?

Reply:
> Why don't we just remove keep_intermediate=True?

I'm afraid that the test author put it there for a purpose?

Reply:
But I think we can remove it. Nothing in the test actually uses the intermediates.

Reply:
Just remove it.

Reply:
done